examples/utils/migration-tools/api-report/migration-tools.alpha.api.md

```diff
@@ -0,0 +1,186 @@
+## Alpha API Report File for "@fluid-example/migration-tools"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+// @alpha
+export type CreateModelCallback<ModelType> = (runtime: IContainerRuntime, container: IContainer) => Promise<ModelType>;
+
+// @alpha
+export type DataTransformationCallback = (exportedData: unknown, modelVersion: string) => Promise<unknown>;
+
+// @alpha
+export interface IAcceptedMigrationDetails {
+    migrationSequenceNumber: number;
+    newVersion: string;
+}
+
+// @alpha
+export interface IAttachedMigratableModel<ModelType> {
+    migrationTool: IMigrationTool;
+    model: ModelType;
+}
+
+// @alpha
+export interface IDetachedMigratableModel<ModelType> {
+    attach: () => Promise<string>;
+    migrationTool: IMigrationTool;
+    model: ModelType;
+}
+
+// @alpha
+export interface IImportExportModel<ImportType, ExportType> {
+    exportData: () => Promise<ExportType>;
+    importData: (initialData: ImportType) => Promise<void>;
+    supportsDataFormat: (initialData: unknown) => initialData is ImportType;
+}
+
+// @alpha
+export interface IMigratableModel extends IVersionedModel, IImportExportModel<unknown, unknown> {
+    dispose(): void;
+}
+
+// @alpha (undocumented)
+export interface IMigratableModelContainerRuntimeEntryPoint<T> {
+    // (undocumented)
+    getModelAndMigrationTool(container: IContainer): Promise<{
+        model: T;
+        migrationTool: IMigrationTool;
+    }>;
+}
+
+// @alpha (undocumented)
+export interface IMigratableModelLoader<ModelType> {
+    createDetached(version: string): Promise<IDetachedMigratableModel<ModelType>>;
+    loadExisting(id: string): Promise<IAttachedMigratableModel<ModelType>>;
+    loadExistingToSequenceNumber(id: string, sequenceNumber: number): Promise<IAttachedMigratableModel<ModelType>>;
+    supportsVersion(version: string): Promise<boolean>;
+}
+
+// @alpha (undocumented)
```

Comment:
I realize these are all existing APIs, and this is just a new report. But there are a lot of undocumented API members here. We should ensure these get documented before we begin advertising this to customers.